src/main/java/io/github/programminglife2016/pl1_2016/parser/nodes/NodeCollectionSerializer.java

```diff
@@ -28,29 +31,33 @@
     public JsonElement serialize(NodeCollection nodeCollection, Type type,
                                  JsonSerializationContext jsonSerializationContext) {
         final Gson nodeBuilder = new GsonBuilder()
-                .registerTypeAdapter(Node.class, new NodeSerializer()).create();
+                .registerTypeAdapter(Segment.class, new NodeSerializer())
+                .registerTypeAdapter(Bubble.class, new NodeSerializer()).create();
         JsonObject jsonObject = new JsonObject();
         jsonObject.add("status", new JsonPrimitive("success"));
         JsonArray nodes = new JsonArray();
         nodeCollection.values().stream()
                 .map(nodeBuilder::toJsonTree)
                 .forEach(nodes::add);
         jsonObject.add("nodes", nodes);
-        JsonArray edges = new JsonArray();
```

Comment:
There are too many lines in this method. The for-loop for example can be defined in a different method.